### PR TITLE
Support orientation locking in SwiftUI views

### PR DESCRIPTION
**Pull request by mpangburn**
iOS 13's default modal presentation style does not respect the `supportedInterfaceOrientations` property of the presented controller.
As a workound, allow a view's lifetime to manage the application's globally supported orientations.

### CODE REVIEW

File: LoopKitUI/Extensions/OrientationLock.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+/// A class whose lifetime defines the application's supported interface orientations.
+///
+/// Construct an OrientationLock as a `@State` property in a SwiftUI view to constrain the view's supported orientations.
+/// To function, `OrientationLock.deviceOrientationController` must be assigned prior to use.
+public final class OrientationLock {
+    private let originalSupportedInterfaceOrientations: UIInterfaceOrientationMask
+
+    /// The global controller for device orientation.
+    /// The property must be assigned prior to instantiating any OrientationLock.
+    public static weak var deviceOrientationController: OrientationLockable!
```
Comment (mpangburn) [inline LoopKitUI/Extensions/OrientationLock.swift]: In Loop, this hook is a reference to the app delegate, which can control the app's supported orientations globally.

Comment (mpangburn) [reply LoopKitUI/Extensions/OrientationLock.swift]: See https://github.com/tidepool-org/Loop/pull/180

File: LoopKitUI/Extensions/OrientationLock.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    public init(_ supportedInterfaceOrientations: UIInterfaceOrientationMask) {
+        originalSupportedInterfaceOrientations = Self.deviceOrientationController.supportedInterfaceOrientations
+        Self.deviceOrientationController.supportedInterfaceOrientations = supportedInterfaceOrientations
+    }
+
+    deinit {
+        Self.deviceOrientationController.supportedInterfaceOrientations = originalSupportedInterfaceOrientations
+    }
```
Comment (mpangburn) [inline LoopKitUI/Extensions/OrientationLock.swift]: Ideally, this work could be done in a ViewModifier using `onAppear` and `onDisappear`—but as of Xcode 11.6 and iOS 13, `onDisappear` isn't called, so we can use an object's lifetime instead.

**Comment by mpangburn** [comment]
Made some API updates so this reads nicely as `.supportedInterfaceOrientations(...)`; I think it's good to go now.

**Review by rickpasetto — APPROVED** [review]
LGTM!  Nice sleuthing!  🏁

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**